src/Idea.hs

```diff
@@ -79,22 +78,28 @@ showEx tt Idea{..} = unlines $
     f "Found" (Just ideaFrom') ++ f "Perhaps" ideaTo' ++
     ["Note: " ++ n | let n = showNotes ideaNote, n /= ""]
     where
-        (ideaFrom', ideaTo') = maybe (ideaFrom, Nothing) (second Just.(reduceJunk ideaFrom)) ideaTo
+        (ideaFrom', ideaTo') = maybe (ideaFrom, Nothing) ((second Just) . reduceJunk . (ideaFrom, )) ideaTo
         f msg Nothing = []
         f msg (Just x) | null xs = [msg ++ " you should remove it."]
                        | otherwise = (msg ++ ":") : map ("  "++) xs
             where xs = lines $ tt x
 
-reduceJunk :: String -> String -> (String, String)
-reduceJunk from to =
-    let
-        (tup, rev_tup) = (***) (both lines) (both (reverse.lines)) (dupe (from, to))
+-- traverse through hints output from both direction to take out all the excess 
+reduceJunk :: (String, String) -> (String, String)
+reduceJunk x@(from, to) =
+    let 
+        tup = both lines x; rev_tup = both (reverse. lines) x
+        -- compare each line to get the first dissimilar position
```

Comment:
This `go` function returns the length of the longest common prefix? If so, please change its name to something that reminds reader of LCP.